app/src/main/java/de/danoeh/antennapod/activity/OnlineFeedViewActivity.java

```diff
@@ -416,14 +432,68 @@ private void showFeedInformation(final Feed feed, Map<String, String> alternateF
 
         viewBinding.backgroundImage.setColorFilter(new LightingColorFilter(0xff828282, 0x000000));
 
-        View header = View.inflate(this, R.layout.onlinefeedview_header, null);
-
-        viewBinding.listView.addHeaderView(header);
+        viewBinding.listView.addHeaderView(headerBinding.getRoot());
         viewBinding.listView.setSelector(android.R.color.transparent);
         viewBinding.listView.setAdapter(new FeedItemlistDescriptionAdapter(this, 0, feed.getItems()));
 
-        TextView description = header.findViewById(R.id.txtvDescription);
+        final int characterLimit = 165;
+        boolean isLong = false;
 
+
+        headerBinding.txtvDescription.setMaxEms(characterLimit);
+        SpannableString originalDesc = null;
+
+        isLong = feed.getDescription().length() > characterLimit;
+
+        if (isLong) {
+            originalDesc = new SpannableString(HtmlToPlainText.getPlainText(feed.getDescription())
+                    .substring(0, characterLimit));
+            originalDesc.setSpan(new ForegroundColorSpan(
+                            headerBinding.txtvDescription.getTextColors().getDefaultColor()), 0,
+                    characterLimit, 0);
+        } else {
+            originalDesc = new SpannableString(HtmlToPlainText.getPlainText(feed.getDescription()));
+            originalDesc.setSpan(new ForegroundColorSpan(
+                    headerBinding.txtvDescription.getTextColors().getDefaultColor()),
+                    0, feed.getDescription().length(), 0);
+        }
+
+        SpannableString readMore = new SpannableString(
+                Html.fromHtml("&#160;" + "  <b>Read more</b>"));
+        readMore.setSpan(new ForegroundColorSpan(R.attr.colorPrimary), 0, readMore.length(), 0);
+
+        SpannableStringBuilder builder = new SpannableStringBuilder();
```

Comment:
This is only needed if `isLong`. You can simplify the code a lot here by moving everything into the `if` branch

Reply:
got it, :)